src/Compilers/CSharp/Portable/Parser/LanguageParser.cs

```diff
@@ -3193,6 +3346,22 @@ private MemberDeclarationSyntax ParseMemberDeclarationCore(SyntaxKind parentKind
             }
         }
 
+        private bool IsExtensionContainerStart()
+        {
+            switch (this.CurrentToken.Kind)
+            {
+                case SyntaxKind.IdentifierToken:
+                    if (CurrentToken.ContextualKind == SyntaxKind.ExtensionKeyword)
+                    {
+                        return IsFeatureEnabled(MessageID.IDS_FeatureExtensions);
+                    }
```

Comment:
do you want to look for the more explicit `extension(` ?

Reply:
We're going to reserve the `extension` keyword (types with that name will be errors and need to be escaped), so I don't see much point to additional subtlety here. Simple seems better.

Reply:
wfm.